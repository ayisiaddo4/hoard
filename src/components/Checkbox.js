import React from 'react';
import PropTypes from 'prop-types';
import {
  Platform,
  View,
  TouchableHighlight,
  ViewPropTypes,
} from 'react-native';
import Icon from 'components/Icon';

const PLATFORM_ICONS = {
  ios: {
    checked: 'ios-checkbox-outline',
    unchecked: 'ios-square-outline',
  },
  android: {
    checked: 'md-checkbox-outline',
    unchecked: 'md-square-outline-blank',
  },
}[Platform.OS];

export default function Checkbox(props) {
  const icon = props.value ? PLATFORM_ICONS.checked : PLATFORM_ICONS.unchecked;
  return (
    <TouchableHighlight onPress={props.onPress}>
      <View style={props.style}>
        <Icon style={props.iconStyle} icon={icon} />
      </View>
    </TouchableHighlight>
  );
}

Checkbox.propTypes = {
  iconStyle: ViewPropTypes.style,
  style: ViewPropTypes.style,
  value: PropTypes.bool,
  onPress: PropTypes.func.isRequired,
};
