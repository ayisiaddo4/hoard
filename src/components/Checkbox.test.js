/* eslint-env jest */

import React from 'react';
import Checkbox from './Checkbox';

import renderer from 'react-test-renderer';

const onPressEvent = jest.fn();

test('renders correctly', () => {
  const tree = renderer.create(<Checkbox onPress={onPressEvent} />).toJSON();
  expect(tree).toMatchSnapshot();
});
