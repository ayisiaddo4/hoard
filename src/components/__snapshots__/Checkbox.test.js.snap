// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={null}
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    style={null}
  >
    <Text
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#FFFFFF",
            "fontSize": 40,
          },
          undefined,
          Object {
            "fontFamily": "Ionicons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
        ]
      }
    >
      ïˆ™
    </Text>
  </View>
</View>
`;
